### PR TITLE
Merge `series/0.23` into `main`

### PR DESCRIPTION
**Pull request by danicheg**
That wasn't a pleasure cruise, so check this intently, please. Especially changes about `FileService`.

### CODE REVIEW

**Review by rossabaker — APPROVED** [review]

File: server/shared/src/main/scala/org/http4s/server/staticcontent/FileService.scala [hunk 1]
```diff
@@ -32,93 +32,158 @@ import org.http4s.server.middleware.TranslateUri
...
 import scala.util.control.NoStackTrace
 
 object FileService {
   private[this] val logger = getLogger
 
-  type PathCollector[F[_]] = (Path, Config[F], Request[F]) => OptionT[F, Response[F]]
+  @deprecated("use FileService.Fs2PathCollector", "0.23.8")
+  type PathCollector[F[_]] = (File, Config[F], Request[F]) => OptionT[F, Response[F]]
+
+  type Fs2PathCollector[F[_]] = (Path, Config[F], Request[F]) => OptionT[F, Response[F]]
```
Comment (armanbilge) [inline server/shared/src/main/scala/org/http4s/server/staticcontent/FileService.scala:44]: This was correct before.

File: server/shared/src/main/scala/org/http4s/server/staticcontent/FileService.scala [hunk 2]
```diff
@@ -32,93 +32,158 @@ import org.http4s.server.middleware.TranslateUri
...
+      "Config is no longer a case class. The productElement method is provided for binary compatibility.",
+      "0.23.8",
+    )
+    override def productElement(n: Int): Any = n match {
+      case 0 => systemPath
+      case 1 => fs2PathCollector
+      case 2 => pathPrefix
+      case 3 => bufferSize
+      case 4 => cacheStrategy
+    }
```
Comment (armanbilge) [inline server/shared/src/main/scala/org/http4s/server/staticcontent/FileService.scala:80]: These should have been removed.

**Review by armanbilge — CHANGES_REQUESTED** [review]
I think `FileService` regressed. Maybe it would be better to keep what's on main and replay https://github.com/http4s/http4s/pull/5837.

File: ember-server/jvm/src/test/scala/org/http4s/ember/server/EmberServerMtlsSuite.scala [hunk 3]
```diff
@@ -125,6 +126,7 @@ class EmberServerMtlsSuite extends Http4sSuite {
       tlsContext <- authTlsClientContext
       emberServer <- EmberServerBuilder
         .default[IO]
+        .withPort(ip4s.Port.fromInt(0).get)
```
Comment (bplommer) [inline ember-server/jvm/src/test/scala/org/http4s/ember/server/EmberServerMtlsSuite.scala:129]: Not a blocker but would be nice to have a string interpolator for this

**Review by bplommer — APPROVED** [review]
Approving this but most of the merge conflicts aren't mine so don't give my approval too much weight.

File: ember-server/jvm/src/test/scala/org/http4s/ember/server/EmberServerMtlsSuite.scala (hunk 3 above)
Comment (armanbilge) [reply ember-server/jvm/src/test/scala/org/http4s/ember/server/EmberServerMtlsSuite.scala:129]: Ah, that's on me, sorry. We should upstream that change from 0.23 actually.

Comment (bplommer) [reply ember-server/jvm/src/test/scala/org/http4s/ember/server/EmberServerMtlsSuite.scala:129]: No worries, this wasn't really the place to bring it up anyway

File: server/shared/src/main/scala/org/http4s/server/staticcontent/FileService.scala (hunk 1 above)
Comment (danicheg) [reply server/shared/src/main/scala/org/http4s/server/staticcontent/FileService.scala]: Thanks! I felt that something was wrong.

**Review by armanbilge — APPROVED** [review]
Thanks for fixing ❤️

**Pull request merged**